ldap-client.go

```diff
@@ -244,3 +247,89 @@ func (lc *LDAPClient) GetAllGroupsByName(groupName string) ([]LdapGroup, error)
 	}
 	return groups, nil
 }
+
+// ChangeUserPassword changes user's password.
+// Note - currently this function is relevant only to Active Directory.
+func (lc *LDAPClient) ChangeUserPassword(username, oldPassword, newPassword string) (err error) {
+	err = lc.Connect()
+	if err != nil {
+		return
+	}
+
+	// first bind with a read only user
+	if lc.BindDN != "" && lc.BindPassword != "" {
+		if err = lc.Conn.Bind(lc.BindDN, lc.BindPassword); err != nil {
+			err = fmt.Errorf("could not bind read only user: %s", err.Error())
+			return
+		}
+	}
+
+	attributes := append(lc.Attributes, "dn")
+	// Search for the given username
+	searchRequest := ldap.NewSearchRequest(
+		lc.Base,
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		fmt.Sprintf(lc.UserFilter, username),
+		attributes,
+		nil,
+	)
+
+	sr, err := lc.Conn.Search(searchRequest)
+	if err != nil {
+		return err
+	}
+
+	if len(sr.Entries) < 1 {
+		return errUserNotExist
+	}
+
+	if len(sr.Entries) > 1 {
+		return errTooManyEntries
+	}
+
+	userDN := sr.Entries[0].DN
+	user := map[string][]string{}
+	for _, attr := range lc.Attributes {
+		user[attr] = sr.Entries[0].GetAttributeValues(attr)
+	}
+
+	// bind as the user to verify their current password
+	if err = lc.Conn.Bind(userDN, oldPassword); err != nil {
+		err = fmt.Errorf("could not bind user via old password: %s", err.Error())
+		return
+	}
+
+	utf16 := unicode.UTF16(unicode.LittleEndian, unicode.IgnoreBOM)
+
+	var encodePasswordForAD = func(pass string) (encoded string, err error) {
+		if encoded, err = utf16.NewEncoder().String(fmt.Sprintf("\"%s\"", pass)); err != nil {
+			err = fmt.Errorf("password could not be utf16 encoded: %s", err.Error())
```

Comment:
wrapping is better than simply flattening the error 
```suggestion
			err = fmt.Errorf("password could not be utf16 encoded: %w", err)
```